model/security/role.smithy

```diff
@@ -16,8 +16,8 @@ structure Role {
     reserved: Boolean
     hidden: Boolean
     description: String
-    cluster_permission: ClusterPermission
-    index_permission: IndexPermission
+    cluster_permissions: ClusterPermission
+    index_permissions: IndexPermissionList
     tenant_permissions: TenantPermission
```

Comment:
`tenant_permissions` should also actually be a list of objects, rather than a list of strings. With each `TenantPermission` being of shape (in pseudo-code):
```
{
    tenant_patterns: List<String>,
    allowed_actions: List<String>
}
```